### PR TITLE
[Merged by Bors] - feat : add lemma on nilpotent and powers

### PR DESCRIPTION
**Pull request by XavierXarles**
Changed `IsNilpotent.pow to IsNilpotent.pow_succ`. Added `IsNilpotent.of_pow` and `IsNilpotent.pow_iff_pos`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Nilpotent.lean [hunk 1]
```diff
@@ -58,17 +58,28 @@ theorem IsNilpotent.neg [Ring R] (h : IsNilpotent x) : IsNilpotent (-x) := by
...
 
-lemma IsNilpotent.pow {n : ℕ} {S : Type*} [MonoidWithZero S] {x : S}
+lemma IsNilpotent.pow_succ (n : ℕ) {S : Type*} [MonoidWithZero S] {x : S}
     (hx : IsNilpotent x) : IsNilpotent (x ^ n.succ) := by
   obtain ⟨N,hN⟩ := hx
   use N
   rw [← pow_mul, Nat.succ_mul, pow_add, hN, mul_zero]
 
+theorem  IsNilpotent.of_pow [MonoidWithZero R] (x : R) (m : ℕ)
+    (h : IsNilpotent (x ^ m)) : IsNilpotent x := by
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Nilpotent.lean:68]:
```suggestion
theorem  IsNilpotent.of_pow [MonoidWithZero R] {x : R} {m : ℕ}
    (h : IsNilpotent (x ^ m)) : IsNilpotent x := by
```

**Review by riccardobrasca — APPROVED** [review]
The diff says you deleted the `.gitignore` file, can you revert that change?

**Comment by riccardobrasca** [comment]
I've pushed a commit that fixes it. It should be good now. Thanks!

bors d+

**Comment by XavierXarles** [comment]
> ✌️ XavierXarles can now approve this pull request. To approve and merge a pull request, simply reply with `bors r+`. More detailed instructions are available [here](https://bors.tech/documentation/getting-started/#reviewing-pull-requests).

bors r+

**Comment by riccardobrasca** [comment]
There is a build error (due to the fact that the variable is now implicit I guess). Can you fix it please? In general it's a good idea to wait for CI to finish before merging the PR (meaning to wait that all the checks are green).

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**